src/classes/VOL_CTRL_VolunteersJobListingFS.cls

```diff
@@ -270,7 +271,7 @@ global virtual with sharing class VOL_CTRL_VolunteersJobListingFS {
                         Location_Street__c, Location_City__c, Location__c, Location_Zip_Postal_Code__c, External_Signup_Url__c,
                         (Select Id, Start_Date_Time__c, Duration__c, Number_of_Volunteers_Still_Needed__c, Total_Volunteers__c,
                             Description__c, System_Note__c From Volunteer_Job_Slots__r 
-                            where Start_Date_Time__c >= :dtNow and Start_Date_Time__c < :dtLast
+                            //where Start_Date_Time__c >= :dtNow and Start_Date_Time__c < :dtLast
```

Comment:
Returning too many records is a valid concern. One suggestion I have is: put the Start_Date_Time__c < :dtLast condition back, so we don't return any future shifts. In addition to to that, we can also ad a limit to the number of past shifts returned by adding  LIMIT 20, for example. Let me know what you think. Thanks.